### PR TITLE
Enable user registration

### PR DESCRIPTION
**Pull request by xmnlab**
This PR aims to enable the user registration option

### CODE REVIEW

File: docker/superset/superset_config.py [hunk 1]
```diff
@@ -116,4 +116,24 @@ class CeleryConfig:
     # "sk": {"flag": "sk", "name": "Slovak"},
     # "sl": {"flag": "si", "name": "Slovenian"},
     # "nl": {"flag": "nl", "name": "Dutch"},
-}
\ No newline at end of file
+}
+
+# EMAIL
+
+MAIL_SERVER = 'smtp.gmail.com'
```
Comment (fccoelho) [inline docker/superset/superset_config.py:123]: So in the end you opted for gmail?

File: docker/superset/superset_config.py [hunk 2]
```diff
@@ -116,4 +116,24 @@ class CeleryConfig:
...
+MAIL_USE_TLS = True
+MAIL_PORT = 587
+MAIL_USERNAME = 'epigraphhub@thegraphnetwork.org'
+MAIL_PASSWORD = 'vlb_1GqfK7c'
```
Comment (fccoelho) [inline docker/superset/superset_config.py:127]: Should this password be open?

**Review by fccoelho — CHANGES_REQUESTED** [review]

**Comment by fccoelho** [comment]
@xmnlab have you found any mechanism for new account approval?

I am worried that at some point we may need to control the demand for new accounts  in the system, particularly if we can't upgrade the server to handle the extra load.

File: docker/superset/superset_config.py (hunk 1 above)
Comment (xmnlab) [reply docker/superset/superset_config.py]: it should be defined in the .env file
for now I am using the gmail just in order to have it ready as soon as possible
it seems that gmail will change some security options and maybe we will need to check some other alternatives as well

to summarize the free alternatives:
- sendinblue maybe would work .. but it depends on their approval in order to use the service. first time I used it was approved (using my account) but second  time it was declined (using the epigraphhub account).
- mailgun looks easy to configure, but it allows to send message just to pre-registered email (that doesn't work for us)
- mailjet looks nice, but not sure yet about the restrictions after the free trial

**Comment by xmnlab** [comment]
> @xmnlab have you found any mechanism for new account approval?
> 
> I am worried that at some point we may need to control the demand for new accounts in the system, particularly if we can't upgrade the server to handle the extra load.

checking that now: https://github.com/apache/superset/discussions/19922

**Comment by xmnlab** [comment]
@fccoelho I fixed the issues you pointed.

about the extra layer for approving new user creation request. I didn't receive any answers yet.
is it a block?

**Comment by fccoelho** [comment]
No, it's not a block. It is just a nice thing to have in place at some point, a basic policy for account activation and deactivation.
for now, we can take care of that manually.

**Review by fccoelho — APPROVED** [review]

**Comment by xmnlab** [comment]
thanks for the explanation @fccoelho

**Pull request merged**